discord_bots/cogs/economy.py

```diff
@@ -0,0 +1,981 @@
+from typing import Any, Literal, Optional
+from datetime import datetime, timedelta, timezone
+from pytz import utc
+
+from discord import (
+    app_commands,
+    ButtonStyle,
+    Client,
+    Colour,
+    Embed,
+    Interaction,
+    Message,
+    TextChannel,
+    TextStyle,
+    VoiceChannel,
+)
+from discord.member import Member
+from discord.ext.commands import Bot, check, Cog
+from discord.ui import Modal, TextInput, View, Button
+
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.session import Session as SQLAlchemySession
+
+from discord_bots.bot import bot
+from discord_bots.checks import is_admin
+from discord_bots.cogs.base import BaseCog
+from discord_bots.config import ECONOMY_ENABLED, CURRENCY_NAME, PREDICTION_TIMEOUT
+from discord_bots.models import (
+    Player,
+    FinishedGame,
+    InProgressGame,
+    InProgressGameChannel,
+    InProgressGamePlayer,
+    Session,
+    EconomyDonation,
+    EconomyPrediction,
+    EconomyTransaction,
+)
+
+
+class EconomyCommands(BaseCog):
+    def __init__(self, bot: Bot) -> None:
+        super().__init__(bot)
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_ready(self):
```

Comment:
Instead of using a `@Cog.listener()`, move this logic to a `cog_load()` function